### DISCUSSION TITLE
Proposal: `Release when all checks passed` option when submitting a release

### DISCUSSION THREAD

**Discussion by MarvinJWendt** [community/community#43356] [Repositories]
### Select Topic Area

Product Feedback

### Body

Hello, I am the maintainer and owner of several open-source projects. Most, if not all, of my projects have a very rapid release cycle, where each pull request is released immediately after it is merged. However, some of my projects have a longer build and testing time, up to 10 minutes. I have experienced instances where I start a release and then wait for all checks to complete, but occasionally forget to check the status after a few minutes and go offline. This can result in the pull request not being released until I remember and take action.

### Proposal
I would like to suggest the implementation of a feature called "Auto Release when all checks pass". This feature would work similarly to the automatic merging of pull requests, and would allow me to prepare and submit release notes in advance. If any checks fail, I will receive a notification and can manually resubmit the release after addressing the issue. This will help prevent accidental releases with outdated changelogs after bug fixes. If all checks pass, the release will be automatically published.

For security purposes, newly added commits should not be included in a release while it is waiting for all checks to pass. The release should be based on the last commit made at the time the release was initiated.

**Comment by spenserblack** [comment]
Until this is implemented, one thing I might consider is to automate the creation of releases on a push to `master`/`main`/`trunk`/etc. with a GitHub Action. By automating release creation you can add checks that must pass before the release is created.

You could possibly give your merge commits titles like `MAJOR: Overhaul foo` or `PATCH: Fix bar` to hint to the automation what the new version should be, and grab release notes from the "Unreleased" section of your changelog.

You could also add your release notes as files in your repo. If, for example, it's called `releases/v1.2.3.txt`, you can detect when a new file is added to `releases/` and create the release from its filename and contents.

Depending on how you generate release notes, even a `workflow_dispatch` event could work for you.
